sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java

```diff
@@ -2492,6 +2507,69 @@ interface WithSecurityFeatures {
              */
             Update withoutVTpm();
         }
+
+        /** The stage of the VM update allowing to change delete options of resources attached to this VM . */
+        interface WithDeleteOptions {
+            /**
+             * Specifies delete options for the OS disk of the VM.
+             *
+             * @param deleteOptions delete options for the OS disk
+             * @return the next stage of the update
+             */
+            Update withOsDiskDeleteOptions(DeleteOptions deleteOptions);
+
+            /**
+             * Specifies delete options for the primary network interface of the VM.
+             *
+             * @param deleteOptions delete options for the primary network interface
+             * @return the next stage of the update
+             */
+            Update withPrimaryNetworkInterfaceDeleteOptions(DeleteOptions deleteOptions);
+
+            /**
+             * Specifies delete options for the secondary network interfaces.
+             * It also sets the default delete options for the secondary network interfaces.
+             *
+             * @param deleteOptions delete options for the secondary network interfaces
+             * @return the next stage of the update
+             */
+            Update withSecondaryNetworkInterfacesDeleteOptions(DeleteOptions deleteOptions);
+
+            /**
+             * Specifies delete options for the network interface.
+             *
+             * @param networkInterfaceId resource ID of the network interface attached to the VM
+             * @param deleteOptions delete options for the network interface
+             * @return the next stage of the update
+             */
+            Update withNetworkInterfaceDeleteOptions(String networkInterfaceId, DeleteOptions deleteOptions);
+
+            /**
+             * Specifies delete options for the network interfaces attached to the VM.
+             *
+             * @param deleteOptions delete options for the network interfaces
+             * @return the next stage of the update
+             */
+            Update withNetworkInterfacesDeleteOptions(DeleteOptions deleteOptions);
```

Comment:
Does it set for default as well?

Reply:
The 3 "modify exists + modify new + set default" API could potentially be confusing.

```java
vm.update()
  .withNewDataDisk(disk1, DeleteOptions.DETACH)
  .withDataDisksDeleteOptions(DeleteOptions.DELETE)
  .withNewDataDisk(disk2, DeleteOptions.DETACH)
```

It would be ambiguous why `withDataDisksDeleteOptions(DeleteOptions.DELETE)` should change disk1.

Reply:
According to current logic, disk1 and disk2 will all be changed to `DeleteOptions.DELETE`. 

I also feel the batch update methods are a bit too much. I feel like changing them to `withDeleteOptionsForExistingXXs()` and limit to existing ones.  Seems much clearer to the user. What do you think?

Reply:
I actually didn't expect disk2 be `DeleteOptions.DELETE` in current code, as it after the line take effect. If so, that is more confusing than I had thought.

`WithExisting` may still not clear. If I call it in the last line, the added disk above is counted as "existing" or nor? That is open to interpretation.

Reply:
But I don't have better idea/name.

"modify exists" at least would be much simpler than the 3-in-1

Reply:
Oh, didn't see `DeleteOptions.DETACH` of the last new data disk. It'll be `DETACHED` and will not change. So it should be disk1->DELETE and disk2-> DETACH at the end..

Reply:
@haolingdong-msft Opinion?

Reply:
Or we cut off the support for batch update.

Reply:
Yeah, that is also an option. User can still do it, but less convenient (and hard to do programmingly, as user need to first get how many disk there is, and `withDeleteOptions` on all of them).

Reply:
We can probably use `List` to as input parameter for `withDataDisksDeleteOptions`? Like below will set delete options for disk 1,2,3. This could be less confusing.
```java
 .withDataDisksDeleteOptions([1,2,3], DeleteOptions.DELETE)
```

Reply:
Haoling's idea is to modify the batch update's method signature to `withDataDisksDeleteOptions(List<Integer> luns, DeleteOptions options)` so that user can explicitly input disks to update. In reality, it would be 
```java
vm.update()
.withNewDataDisk(1)
.withDataDisksDeleteOptions(new ArrayList<>(vm.dataDisks().keySet()), DeleteOptions.DELETE)
.withDataDiskDeleteOptions(1, DeleteOptions.DELETE)
```

Same for network interfaces. Probably no update for secondary, just `withNetworkInterfacesDeleteOptions(List<String> nicId, DeleteOptions deleteOptions)`, and user can call:
```java
vm.update()
.withNetworkInterfaceDeleteOptions(vm.secondaryNetworkInterfaces()) // need to provide new method `vm.secondaryNetworkInterfaces()`
```

@haolingdong-msft correct me if I'm wrong.

Reply:
I think that would work too.

We may not exactly need a `secondaryNetworkInterfaces()`. What the List does here is that it would avoid user to decide how many time to call `withDataDisksDeleteOptions(int, ..)` which is hard to do in fluent API.

Reply:
Or you may reverse the parameter order and use vargs.

Reply:
Sure, just saw `networkInterfaceIds()` in `VirtualMachine`. That'll do.

Reply:
@XiaofeiCao, Yes, the idea is to let use explicitly define which data disks to modify intead of us guessing what they want to do. Also the user can call `withDataDisksDeleteOptions` once to set all the data disk options.
BTW, To me,  providing one API `withDataDisksDeleteOptions(List<Integer> luns, DeleteOptions options)` would be enough. @XiaofeiCao @weidongxu-microsoft What do you think? (I'm flexible about this, it's fine if you feel two methods would be more convenient for user.)

Reply:
Yeah, one interface seems enough for me. We can use varargs to make it more approachable for single update.

Reply:
Sure, varargs works!